conda_smithy/linter/hints.py

```diff
@@ -28,6 +29,25 @@ def hint_pip_usage(build_section, hints):
                 )
 
 
+def hint_sources_should_not_mention_pypi_io_but_pypi_org(
+    sources_section: List[Dict[str, Any]], hints: List[str]
+):
+    """
+    Grayskull and conda-forge default recipe used to have pypi.io as a default,
+    but cannonical url is PyPI.org.
+
+    See https://github.com/conda-forge/staged-recipes/pull/27946
+    """
+    for source_section in sources_section:
+        if (source_section.get("url", None) is not None) and source_section[
+            "url"
+        ].startswith("https://pypi.io/"):
```

Comment:
```suggestion
        if source_section.get("url", "").startswith("https://pypi.io/"):
```
I think this is the same?

Reply:
Yeah it is not the same because if url is set to None, it still returns None.